### ISSUE TITLE
feat: All Day Events

### ISSUE THREAD

**Issue by eiclu** [calcom/cal.com#11499]
### Is your proposal related to a problem?

Right now it is not possible to allow people to simply book a whole day.

### Describe the solution you'd like

When an appointment spans all available time in a day, it should get selected immediately without opening the time selection. Maybe add a "whole day" option to event durations.

### Describe alternatives you've considered

It is possible to extend the working hours for each day to 0:00 - 23:59, but on the customer side it looks awkward with one slot available each day:

![image](https://github.com/calcom/cal.com/assets/6774795/42455804-a13b-4c94-a2a6-de8fe362a97c)

**Comment by PeerRich** [comment]
we thought about this in the past but were not able to convince ourselves yet to prioritize this higher. will talk to the team. more activity in this ticket would help to see if this is valuable

**Comment by silveltman** [comment]
I would love this feature. It would open up the possibility to use Cal.com for clients as an appointment scheduling tool for their businesses.

Let's say you have a repair business at home, these companies don't want the customer to choose the time, since they want to decide the time themselves based on the route they have through the day.

**Comment by eiclu** [comment]
This would open completely new Use Cases for cal.com. Examples include:

* Hotel Room Booking
* Car Rental
* Equipment Lending

**Comment by silveltman** [comment]
A relatively simple solution would be a setting where we could choose to also show the end-time, instead of just the starting-time of an event. Something like this:

![Screen Arc 18-12-2024 12 17 47](https://github.com/user-attachments/assets/2e47e7aa-a0b7-401e-b4fa-74bc5c7eb8a5)

I think this would even make sense in more cases. The label that indicates the duration of an event is very very small, there are probably people not seeing this.

**Comment by SLCPereira** [comment]
Exactly. I actually signed up with the idea of integrating Cal.com into a website for a short-term rental. This would be super helpful.

**Comment by PeerRich** [comment]
continuing in #20441
